theano/tensor/blas.py

```diff
@@ -304,14 +305,16 @@ def default_blas_ldflags():
                 ['-L%s' % l for l in blas_info['library_dirs']] +
                 ['-l%s' % l for l in blas_info['libraries']] +
                 [])
-            if GCC_compiler.try_flags(ret):
+            if GCC_compiler.try_flags(ret) and try_blas_flag():
```

Comment:
Doing those 2 tests are do duplicate test. Just call try_blas_flag, is cover what try_flags was doing.

But modify try_blas_flag() to take ret as a parameter and use it.